res/poketool/pokegra/pl_pokegra/meson.build

```diff
@@ -0,0 +1,92 @@
+pl_pokegra_srcs = []
+pl_pokegra_files = []
+
+sprites_flags = ['-scanfronttoback', '-handleempty']
+palette_flags = ['-bitdepth', '8', '-nopad', '-comp', '10']
+
+foreach i : range(494)
+    padded_i = run_command('printf', '%04d', i.to_string(), check: false).stdout()
+
+    female_back = files(padded_i / 'female' / 'back.png')
+    male_back = files(padded_i / 'male' / 'back.png')
+    female_front = files(padded_i / 'female' / 'front.png')
+    male_front = files(padded_i / 'male' / 'front.png')
+    pl_pokegra_srcs += [female_back, male_back, female_front, male_front]
+
+    pl_pokegra_files += custom_target(padded_i + '-00.NCGR',
+        output: padded_i + '-00.NCGR',
+        input: female_back,
+        command: [
+            nitrogfx_exe,
+            '@INPUT@',
+            '@OUTPUT@',
+            sprites_flags
+        ]
+    )
+    pl_pokegra_files += custom_target(padded_i + '-01.NCGR',
+        output: padded_i + '-01.NCGR',
+        input: male_back,
+        command: [
+            nitrogfx_exe,
+            '@INPUT@',
+            '@OUTPUT@',
+            sprites_flags
+        ]
+    )
+    pl_pokegra_files += custom_target(padded_i + '-02.NCGR',
+        output: padded_i + '-02.NCGR',
+        input: female_front,
+        command: [
+            nitrogfx_exe,
+            '@INPUT@',
+            '@OUTPUT@',
+            sprites_flags
+        ]
+    )
+    pl_pokegra_files += custom_target(padded_i + '-03.NCGR',
+        output: padded_i + '-03.NCGR',
+        input: male_front,
+        command: [
+            nitrogfx_exe,
+            '@INPUT@',
+            '@OUTPUT@',
+            sprites_flags
+        ]
+    )
+
+    if i == 0
+        # nitrogfx doesn't like these two files, so copy over the canonical versions
+        pl_pokegra_files += fs.copyfile('0000/0000-04.NCLR', '0000-04.NCLR')
+        pl_pokegra_files += fs.copyfile('0000/0000-05.NCLR', '0000-05.NCLR')
+        continue
+    endif
+
+    if run_command('test', '-s', male_front, check: false).returncode() == 0
```

Comment:
Use the filesystem module instead
https://mesonbuild.com/Fs-module.html#is_symlink

Reply:
Does that work? The files *do* exist in these cases, they just have size 0.

Reply:
Oh, in that case https://mesonbuild.com/Fs-module.html#size
I'm not entirely sure what the check is for, anyhow.

Reply:
It's just to make sure we're checking a file that actually has data for compiling the palette out of it. Ungendered mons have their sprites stored as `male`, so it's really just to check if the 'mon in question is female-only (e.g., Nidoqueen).

Reply:
Why does the male sprite exist at all, then?

Reply:
Because the NARC mandates it. Vanilla has the same structure, having 0-length files for non-existent gender-forms.

Reply:
In that case I'd create empty files with meson if the source doesn't exist. You can loop through a {male,female,front,back} array to simplify the logic.

Reply:
Instead of storing empty PNGs I mean

Reply:
> In that case I'd create empty files with meson if the source doesn't exist. You can loop through a {male,female,front,back} array to simplify the logic.

That's fair. I'll implement that tomorrow and clean up the original dump commit.